barcodes_generator_partner/views/view_res_partner.xml

```diff
@@ -7,12 +7,12 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 -->
 
 <odoo>
-
     <record id="view_res_partner_form" model="ir.ui.view">
         <field name="model">res.partner</field>
-        <field name="inherit_id" ref="point_of_sale.view_partner_property_form"/>
+        <field name="inherit_id" ref="base.view_partner_form"/>
         <field name="arch" type="xml">
-            <field name="barcode" position="before">
+            <xpath expr="//field[@name='industry_id']" position="after">
+                <field name="barcode"/>
```

Comment:
This line is new.

Reply:
So please explain howto Add this functionality to Partner without installing point of sale👌

Reply:
@legalsylvain IMHO @agenterp remark is correct. The dependency on point of sale should be avoided

Reply:
> The dependency on point of sale should be avoided

I'm fully agree ! But it is not a matter of OCA, but a matter of Odoo, and should be fixed in the core. For the time being, barcode field is defined in the module ``point_of_sale``.
https://github.com/odoo/odoo/blob/master/addons/point_of_sale/models/res_partner.py#L16 (*)

This PR is breaking Odoo partner form. Step to reproduce : 
- install this module barcodes_generator_abstract + point_of_sale : 
- Go to partner form 
- The partner form has two field barcodes.

![image](https://user-images.githubusercontent.com/3407482/70514134-48fc5580-1b33-11ea-8b88-d6bea7337a7d.png)

**Note :** 
For V12, the barcode field is in base module ([in the model](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_partner.py#L172)) but was in the point_of_sale module (for the view). This point was fixed by https://github.com/odoo/odoo/pull/25070/files 
- For V13, and master, this move has been removed, and the field is again in point_of_sale module. 

@mart-e : Do you know why ``barcode`` partner field has moved into base in V12, and has moved again in ``point_of_sale`` for V13 and master ? 

Thanks !

Reply:
@legalsylvain i agree that it's not nice. but we cannot rely on them to fix it.
so how can we provide a barcode to partner without installing pos?

Reply:
@agenterp The solution I've already done in another place is to get this with possible duplicate (don't add dependency here).

And in pos repository add a module 'auto-install' with this module + pos in dependencies to remove the duplicate in view. That's not ideal, but we need to make it change from master in odoo (I commented the PR)

Reply:
Well, this is not acceptable in a stable revision for all instances that are using this module in production, this patch will break form views and the patch will not be installed. (if pos repository is not present, is not updated, if installation is done by pip, etc...)

- I propose to close this PR, and work with @mart-e for master revision. (maybe it will not be necessary)
- If interested, please use this branch as a patch in production.
- for the report part, I think that it should be introduced in a module ``barcodes_partner_report`` and not in this one, because it has nothing to do with the generation of the barcode.

regards.